cover-scheduled-event-handlers.js

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021 Allterco Robotics EOOD
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Shelly is a Trademark of Allterco Robotics
+
+// Shelly Script example: Control a open or close a Shelly Plus 2PM (Gen2) on cover mode
+// by handling specific events
+//
+// The script, when run, will subscribe to events and handle "cover_open_to" and
+// "cover_close_to" events.
+//
+// The script requires the event data to specify the desired position and optionally
+// the hours and minutes of the day which the event should not be executed before.
+//
+// Example event data:
+// {
+//   "pos": 100,
+//   "not_before": {
+//     "h": 8,
+//     "m": 0
+//   }
+// }
+//
+// Note: Currently, "not_before" is only supported for "cover_open_to" events.
+//
+// The event can be published on a schedule as follows:
+//
+// {
+//   "id": 8,
+//   "enable": true,
+//   "timespec": "@sunrise+00h00m * * MON,TUE,WED,THU,FRI",
+//   "calls": [
+//       {
+//           "method": "script.eval",
+//           "params": {
+//               "id": 1,
+//               "code": "Shelly.emitEvent(\"cover_open_to\",{\"pos\":60,\"not_before\":{\"h\":7,\"m\":30})"
```

Comment:
Good work! Just a suggestion: you can define 2 functions something like CoverOpenTo(pos, hour, minute) and CoverCloseTo, that will emit the event. These functions can be used with schedules, because its a bit hard for everyone to escape the string like this. Its easier to have code: "CoverOpenTo(60, 7, 30)"

Reply:
The idea here was precisely not to do that and decouple the event from publishing from event handling.

The schedules UI could have an option to publish custom events. 😄